pilot/pkg/security/authz/matcher/header.go

```diff
@@ -101,36 +96,21 @@ func HostMatcher(k, v string) *routepb.HeaderMatcher {
 		return &routepb.HeaderMatcher{
 			Name: k,
 			HeaderMatchSpecifier: &routepb.HeaderMatcher_StringMatch{
-				StringMatch: &matcher.StringMatcher{
-					IgnoreCase: true,
-					MatchPattern: &matcher.StringMatcher_Suffix{
-						Suffix: v[1:],
-					},
-				},
+				StringMatch: StringMatcherSuffix(strings.TrimPrefix(v, "*"), true),
```

Comment:
Why is this ignoreCase=true here but in the YAML I don't see anything with `ignoreCase: true`?

Reply:
How is it that when you changed this from `true` to `false` no tests failed? This makes me uncomfortable with our test coverage. Can we expand the tests maybe?

Reply:
Ping on this

Reply:
Since I am not familiar with github operations, the previously replied message has been pending and was sent today.